plugins/git/git_commit_behind

```diff
@@ -0,0 +1,260 @@
+
+conf = {
+    'git_path':  os.getenv('git_path', '/usr/bin/git'),
+    'state_file': os.getenv('MUNIN_STATEFILE',
+                            '/var/lib/munin-node/plugin-state/nobody/' +
+                            'git_commit_behind.state')
+}
+
+repo_codes = set([re.search('repo\.([^.]+)\..*', elem).group(1)
+                  for elem in os.environ.keys() if 'repo.' in elem])
+
+repos_conf = {}
+for code in repo_codes:
+    repos_conf[code] = {
+        'name':        os.getenv('repo.%s.name' % code, code),
+        'path':        os.getenv('repo.%s.path' % code, None),
+        'user':        os.getenv('repo.%s.user' % code, None),
+        'warning':     os.getenv('repo.%s.warning' % code, '10'),
+        'critical':    os.getenv('repo.%s.critical' % code, '100')
+    }
+
+
+def print_config():
+    print('graph_title Git repositories - Commits behind')
+
+    print('graph_args --base 1000 -r --lower-limit 0')
+    print('graph_vlabel number of commits behind')
+    print('graph_scale yes')
+    print('graph_info This graph shows the number of commits behind' +
+          ' for each configured git repository')
+    print('graph_category system')
+
+    print('graph_order %s' % ' '.join(repo_codes))
+
+    for repo_code in repos_conf.keys():
+        print('%s.label %s' % (repo_code, repos_conf[repo_code]['name']))
+        print('%s.warning %s' % (repo_code, repos_conf[repo_code]['warning']))
+        print('%s.critical %s' %
+              (repo_code, repos_conf[repo_code]['critical']))
+
+
+def generate_git_command(repo_conf, git_command):
+    if not repo_conf['user'] or repo_conf['user'] == os.environ['USER']:
+        cmd = [conf['git_path']] + git_command
+    else:
+        shell_cmd = 'cd %s ; %s %s' % (
+            repo_conf['path'], conf['git_path'], ' '.join(git_command))
+        cmd = ['su', '-', repo_conf['user'], '-c', shell_cmd]
+    return cmd
+
+
+def execute_git_command(repo_conf, git_command):
+    cmd = generate_git_command(repo_conf, git_command)
+    return check_output(cmd, cwd=repo_conf['path']).decode('utf-8').rstrip()
+
+
+def get_info():
+    if not os.access(conf['git_path'], os.X_OK):
+        print('Git (%s) is missing, or not executable !' %
+              conf['git_path'], file=sys.stderr)
+        sys.exit(1)
+
+    for repo_code in repos_conf.keys():
+        logging.debug(' - %s' % repo_code)
+        try:
+            remote_branch = execute_git_command(
+                repos_conf[repo_code],
+                ['rev-parse', '--abbrev-ref', '--symbolic-full-name', '@{u}'])
+            logging.debug('remote_branch = %s' % remote_branch)
+
+            commits_behind = execute_git_command(
+                repos_conf[repo_code],
+                ['rev-list', 'HEAD..%s' % remote_branch, '--count'])
+
+            print('%s.value %d' % (repo_code, int(commits_behind)))
+        except CalledProcessError as e:
+            logging.error('Error executing git command : %s' % str(e))
```

Comment:
The logging module supports lazy evaluation for optimizing resource usage: simply use `, e` instead of `% str(e)`.